tests/io/test_tidyxl.py

```diff
@@ -0,0 +1,171 @@
+import pytest
+
+from janitor import io
+from pathlib import Path
+from openpyxl import load_workbook
+from pandas.testing import assert_series_equal
+
+
+TEST_DATA_DIR = "tests/test_data"
+filename = Path(TEST_DATA_DIR).joinpath("worked-examples.xlsx").resolve()
+
+
+wb = load_workbook(filename)
+wb_c = load_workbook(filename, read_only=True)
+
+
+def test_sheetname_type():
+    """Raise error if sheet name is the wrong type."""
+    with pytest.raises(TypeError):
+        io.xlsx_cells(filename, 5)
+
+
+def test_sheetname_presence():
+    """Raise error if sheet name does not exist."""
+    with pytest.raises(KeyError):
+        io.xlsx_cells(filename, "5")
+
+
+def test_comment_read_only():
+    """Raise error if comment and read_only is True."""
+    with pytest.raises(ValueError):
+        io.xlsx_cells(filename, "clean", comment=True)
```

Comment:
For generic errors that we're raising (such as ValueError), let's try to use the `match` wherever possible? Because we're throwing ValueErrors for a lot of different reasons; we don't want to accidentally catch a ValueError from another error msg unknowingly and get a false negative.

Also, I understand `read_only=True` is default set to True, but I think if the test is explicitly meant to check the comment & read_only combination, that we should specify it explicitly in the function call here.